src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -140,7 +142,10 @@ public override IndexOutput CreateOutput(string name, IState s)
             if (state == null)
                 throw new ArgumentNullException(nameof(s));
 
-            return new VoronIndexOutput(_environment.Options, name, state.Transaction, _name);
+            var voronIndexOutput = new VoronIndexOutput(_environment.Options, name, state.Transaction, _name, 
+                beforeFileDispose: () => _voronIndexFiles.Remove(name, out _));
```

Comment:
Maybe instead of creating an action here per file (Lucene tends to create dozen of those). We can create a class `ReleaseFile` and pass one instance of it?

```
public class ReleaseFile
{
    ctor(_voronIndexFiles) { }

   void ReleaseFile(name)
   {
        _voronIndexFiles.Remove(name);
   }
}
```

Reply:
done